include/deal.II/lac/diagonal_matrix.h

```diff
@@ -52,10 +52,16 @@ class DiagonalMatrix : public Subscriptor
   using size_type  = typename VectorType::size_type;
 
   /**
-   * Constructor.
+   * Default constructor. The object needs still to be reinitialized to be
+   * usable.
    */
   DiagonalMatrix() = default;
 
+  /**
+   * Constructor immediately initializing the object properly.
+   */
+  explicit DiagonalMatrix(const VectorType &vec);
```

Comment:
Can you document *what* the object is initialized to? Namely, to a matrix of size nxn where n is the size of the vector, and with diagonal entries equal to the elements of vec.